src/callbacks/execute.py

```diff
@@ -555,3 +664,59 @@ def allow_show_clusters(job_id, project_name):
     except Exception:
         logger.error(traceback.format_exc())
         return True
+
+
+@callback(
+    Output(
+        {
+            "component": "DbcJobManagerAIO",
+            "subcomponent": "run-dropdown",
+            "aio_id": "clustering-jobs",
+        },
+        "options",
+    ),
+    Input(
+        {
+            "component": "DbcJobManagerAIO",
+            "subcomponent": "run-dropdown",
+            "aio_id": "clustering-jobs",
+        },
+        "options",
+    ),
+    prevent_initial_call=True,
+)
+def filter_clustering_dropdown(options):
+    """
+    Filter the clustering job dropdown to only show true clustering jobs.
+    """
+    if not options:
+        return []
+    
+    # Get list of clustering model names for comparison
+    clustering_model_names = [model.lower() for model in clustering_models.modelname_list]
+    
+    # Filter job options
+    filtered_options = []
+    for job_option in options:
+        job_label = job_option.get("label", "").lower()
+        
+        # Check if this job was created by any of the clustering models
+        is_clustering_job = False
+        for model_name in clustering_model_names:
+            if model_name in job_label:
+                is_clustering_job = True
+                break
+        
+        # If no clustering model is mentioned, check if it has clustering-related terms
+        if not is_clustering_job:
+            clustering_terms = ["cluster", "kmeans", "dbscan", "hdbscan", "agglomerative", "hierarchical"]
```

Comment:
Thanks for fixing this dropdown. Could `clustering` be sufficient as a tag to query run flows from? I'm mostly thinking that the options for clustering algorithms may grow in the near future and hence this line may grow - if not, we should move this to an issue to be addressed in a future PR.

Reply:
I think that may not be needed if we add `cluster` to [job_manager_minimal](https://github.com/mlexchange/mlex_utils/blob/6707f921c10419e80858e139b7f8351c1dea728b/mlex_utils/dash_utils/components_bootstrap/job_manager_minimal.py#L101C9-L101C25), but let's explore this in a separate PR - I'm moving this to an issue